### PR TITLE
Support updated_prop

### PR DESCRIPTION
**Pull request by jez**
### Motivation

This is a Stripe-specific prop extension.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
This should also have a sizable drop in missing methods on pay-server.

File: test/testdata/rewriter/prop_updated.rb.rewrite-tree.exp [hunk 1]
```diff
@@ -27,10 +50,14 @@ class <emptyTree><<C <root>>> < (::<todo sym>)
 
     ::Sorbet::Private::Static.keep_self_def(<self>, :"updated_prop")
 
-    <self>.updated_prop()
+    <self>.updated_prop({:"without_accessors" => true})
+
+    ::Sorbet::Private::Static.keep_def(<self>, :"updated")
+
+    ::Sorbet::Private::Static.keep_def(<self>, :"updated=")
```
Comment (jez) [inline test/testdata/rewriter/prop_updated.rb.rewrite-tree.exp:57]: View the individual commits to see the diff to this test before and after.

File: rewriter/Prop.cc [hunk 2]
```diff
@@ -135,17 +135,25 @@ optional<PropInfo> parseProp(core::MutableContext ctx, const ast::Send *send) {
...
+            break;
+        }
         case core::Names::merchantProp()._id:
             ret.isImmutable = true;
             ret.name = core::Names::merchant();
-            ret.nameLoc =
-                core::LocOffsets{send->loc.beginPos(),
-                                 send->loc.endPos() - 5}; // 5 is the difference between `merchant_prop` and `merchant`
+            // 5 is the difference between `merchant_prop` and `merchant`
+            ret.nameLoc = core::LocOffsets{send->loc.beginPos(), send->loc.endPos() - 5};
```
Comment (elliottt) [inline rewriter/Prop.cc:156]: Thanks for making all of these consistent!

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ https://go/builds/bui_HTlJxiVe7PbPxa

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ file
StandaloneBundle
the
can

config
source
rake
ast
bigdecimal
bundler
byebug
coderay
dogstatsd
excon
multipart-post
interception
jaro_winkler
json
little-plugger
multi_json
method_source
mini_portile2
nio4r
parallel
yard
psych
rack
rainbow
rexml
ruby-progressbar
unicode-display_width
rubyzip
subprocess
tins
faraday
logging
nokogiri
parser
pry
sorbet-runtime
term-ansicolor
terminal-table
sentry-raven
pry-byebug
pry-rescue
rubocop
rubocop-daemon
rubocop-minitest
rubocop-performance
pry-doc
sorbet-runtime
complete!
gems
https://go/builds/bui_HTmQ0F7S4J8Tsc

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ https://go/builds/bui_HTnJrjws3KDwai

**Comment by marianosimone** [comment]
Trying to re-surface this, as we've seen this causing issues at Stripe

**Comment by marianosimone** [comment]
Closing and opening a new one, as there are a bunch of merge conflicts

**Pull request closed without merging**